### PR TITLE
Create doc file for chained controllers

### PR DESCRIPTION
**Pull request by destogl**

### CODE REVIEW

**Comment by bmagyar** [comment]
Are you happy with me merging this? People want *some* docs, this is better than nothing!

**Review by bmagyar — CHANGES_REQUESTED** [review]

File: controller_manager/doc/controller_chaining.rst [hunk 1]
```diff
@@ -0,0 +1,102 @@
+Controller Chaining / Cascade Control
+======================================
+
+This document proposes a minimal-viable-implementation of serial controllers chaining as described in [Chaining Controllers design document](controller_chaining.md).
```
[the remark this answers is not in the document]
Comment (christophfroehlich) [reply controller_manager/doc/controller_chaining.rst:4]: I think this one is meant: https://github.com/ros-controls/roadmap/blob/master/design_drafts/controller_chaining.md

File: controller_manager/doc/controller_chaining.rst [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+![Example2](images/chaining_example2.png)
+
+In this example, we want to chain 'position_tracking' controller with 'diff_drive_controller' and two PID controllers.
+Let's now imagine a use case that we don't only want to run all those controllers as a group, but also flexibly add preceding steps.
+This means the following:
+  1. When a robot is stared, we want to check if motor velocity control is working properly and therefore only PID controllers are activated.
+     This means we can control the input of PID controller also externally using topics.
+     But these controllers also provide virtual interfaces, so we can chain them.
+  2. Then "diff_drive_controller" is activated and attach itself to the virtual input interfaces of PID controllers.
```
Comment (christophfroehlich) [inline controller_manager/doc/controller_chaining.rst:28]:
```suggestion
  2. Then "diff_drive_controller" is activated and attaches itself to the virtual input interfaces of PID controllers.
```

**Review by christophfroehlich — CHANGES_REQUESTED** [review]
I hope this gets merged soon ;)

At least a proper rst markup + pictures and link the file to the docs is needed. (Silently advertising https://github.com/ros-controls/control.ros.org/pull/125). 
Or: Push this file as .md file where the suggested changes are accepted, and I could do the translation to rst then.

File: controller_manager/doc/controller_chaining.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (bmagyar) [reply controller_manager/doc/controller_chaining.rst:4]: Thx

**Comment by bmagyar** [comment]
@christophfroehlich could we try the docs on this PR?

**Comment by christophfroehlich** [comment]
this is no valid rst syntax. i'll make an PR against this PR with the changes this weekend

**Comment by bmagyar** [comment]
Ah yeah I'm aware of that :D was just wondering how far it is. But agree that this should be merged asap

**Comment by christophfroehlich** [comment]
I refactored it with PR #1068

**Comment by christophfroehlich** [comment]
@bmagyar this hasn't been backported to humble, which caused the conflicht with #1106. The image and the link in doc/index.rst is missing, see [this error](https://github.com/ros-controls/control.ros.org/actions/runs/6095088775/job/16537947808).

**Comment by christophfroehlich** [comment]
@destogl can this be backported to humble to fix the build error with the docs for humble?

**Comment by destogl** [comment]
We have to fix this. The docs is very wrong and misleading in few places...

**Pull request merged**